File: array_stack.rb
class Stack
  attr_reader :data

  def initialize
    @data = []
  end

  def push(value)
    @data.push(value)
  end

  def pop
    @data.pop
  end

  def empty?
    @data.empty?
  end

  def peek
    @data[-1]
  end
end

File: array_stack.js
function Stack(){
  var data = [];

  this.push = function(element) {
    data.push(element);
  }

  this.pop = function(element) {
    return data.pop();
  }

  this.peek = function() {
    return data[data.length - 1];
  }

  this.isEmpty = function() {
    return data.length == 0;
  }

  this.print = function() {
    console.log(data.toString());
  }
}
